e2e/QUICKSTART.md

```diff
@@ -0,0 +1,243 @@
+
+Let's walk through two quick examples to see how to start working with the e2e-testing framework. We have two similar use cases:
+
+- adding a new test suite
+- working on an existing test suite
+
+## Adding a new test suite
+
+### Step 1 - Install test depedencies
+
+Install the godog binary in the supported version - `make -C e2e install-godog`
+
+### Step 2 - Create a new test suite
+
+Given we create a new test suite **foo** under the "e2e/_suites" workspace. - `SUITE=foo make -C e2e create-suite`
+
+From now on, this is our work directory - `cd e2e/_suites/foo`
+
+### Step 3 - Verify the generated files
+
+We have created a few sample files for you:
+
+- a `foo_test.go` file under the test suite's root directory, including a `TestMain` method representing the test suite. This Go file will represent **the main program** to be executed in the tests, containing a boilerplate structure that includes:
+    - a `FooTestSuite` Go struct representing the state to be passed across steps in the same scenario.
+    - global variables to be passed across steps and scenarios, creating APM traces and spans for the life cycle methods of Godog (before/after Step/Scenario/Suite)
+    - initialisation methods for `godog`'s life cycle hooks: _InitializeFooScenarios_ and _InitializeFooTestSuite_.
+        - **InitializeFooTestSuite**: contains the life cycle hooks for the suite (`BeforeSuite and AfterSuite`)
+        - **InitializeFooScenarios**: contains the life cycle hooks for each test scenario (`BeforeScenario, AfterScenario, BeforeStep and AfterStep`)
+- a `docker-compose.yml` file under `cli/config/compose/profiles/foo`, for the runtime dependencies. This descriptor includes the definition of the services that are needed by our tests before they are run. The sample file contains an Elasticsearch instance, but it could include Kibana, Fleet Server, or any other service in the form of a Docker container.
+- a `foo.feature` feature file under the **features** directory. This directory is the default location for the Gherkin feature files. Although it can be changed to any other location, using the `opts` structure, we recommend keeping it with the default value:
+
+```go
+    // changing the default path is not recommended
+    opts := godog.Options{
+		Paths:     []string{"new-location"},
+	}
+```
+
+You are able now to read the content of the sample feature file - `vim features/foo.feature`
+
+``` gherkin
+Feature: eat godogs
+  In order to be happy
+  As a hungry gopher
+  I need to be able to eat godogs
+
+  Scenario: Eat 5 out of 12
+    Given there are 12 godogs
+    When I eat 5
+    Then there should be 7 remaining
+```
+
+### Step 4 - Create godog step definitions
+
+**NOTE:** same as **go test**, godog respects package level isolation. All your step definitions should be in your test suite root directory. In this case: **foos**.
+
+If we run godog inside the test suite: - `godog`. We can either use Go tools: - `go test -v --godog.format=pretty`
```

Comment:
I think this could be worded a little better and honestly we could probably disregard the use of running `godog` directly and just document using `go test` for everything